agents/meinberg_m1000/meinberg_m1000_agent.py

```diff
@@ -0,0 +1,201 @@
+                for mib in self.mib_timings:
+                    # Select only OIDs of the same interval
+                    if mib['interval'] != interval:
+                        continue
+
+                    if mib["lastGet"] is None:
+                        get_list.append(mib["oid"])
+                    elif time.time() - mib["lastGet"] > mib["interval"]:
+                        get_list.append(mib["oid"])
+
+                if get_list:
+                    # Issue SNMP GET command
+                    result = yield self.snmp.get(get_list)
+                    read_time = time.time()
+
+                if result:
+                    message = {
+                        'block_name': f'm1000_{interval}',
+                        'timestamp': read_time,
+                        'data': {}
+                    }
+
+                    for item in result:
+                        try:
+                            # OID from SNMP GET
+                            oid = item[0].prettyPrint()
+                            # Makes something like 'MBG-SNMP-LTNG-MIB::mbgLtNgRefclockState.1'
+                            # look like 'mbgLtNgRefclockState_1'
+                            valid_field = oid.split("::")[1].replace('.', '_')
+
+                            self.log.debug("{o} {value}",
+                                           o=oid,
+                                           value=int(item[1]))
+                            message['data'][valid_field] = int(item[1])
+                        except ValueError:
+                            self.log.warn('{oid} is of type {_type}, not int',
+                                          oid=item[1], _type=type(item[1]))
+
+                    # Update lastGet time
+                    for mib in self.mib_timings:
+                        if mib['oid'] in get_list:
+                            mib['lastGet'] = read_time
+
+                    self.log.debug("{msg}", msg=message)
+                    session.app.publish_to_feed('m1000', message)
+
+                result = []
+                yield dsleep(0.1)
+
+        return True, "Finished Recording"
+
+    def stop_record(self, session, params=None):
+        """stop_record(params=None)
+
+        Stop method associated with start_record process.
+
+        """
+        self.is_streaming = False
+        return True, "Stopping Recording"
+
+
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group("Agent Options")
+    pgroup.add_argument("--auto-start", default=True, type=bool,
+                        help="Automatically start listening for data at " +
+                        "Agent startup.")
+    pgroup.add_argument("--address", help="Address to listen to.")
```

Comment:
"Listening" is left over from the template agent I think... we are polling, not listening.

Reply:
Yup, good catch, fixed.